### PR TITLE
pkg/ns: consider PROCFS during NS verification

### LINKED ISSUE #226: ns: validation of NS fails on old kernels
The `ns` package will not work properly on older kernels that don't mount network namespaces with the NSFS. This happens on Semaphore CI today.

This is an example failure message:
```
failed to create namespace: /proc/18768/task/18798/ns/net is not of type NS
```

Reproduction of the problem using `ip`
```
Semaphore: ~/rkt $ sudo ip netns add newns; mount | grep newns
proc on /run/netns/newns type proc (rw,nosuid,nodev,noexec,relatime)
$ uname -a
Linux semaphore-1605 3.13.0-32-generic #57-Ubuntu SMP Tue Jul 15 03:51:08 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

### PR DESCRIPTION
**Pull request by steveeJ**
/cc @dcbw 

Fixes #226.

### CODE REVIEW

**Comment by dcbw** [comment]
LGTM though maybe add a commit message talking about how < 3.19 and >= 3.19 handle things here.

File: pkg/ns/ns.go [hunk 1]
```diff
@@ -74,36 +75,55 @@ func GetCurrentNS() (NetNS, error) {
 	return GetNS(getCurrentThreadNetNSPath())
 }
 
-// Returns an object representing the namespace referred to by @path
-func GetNS(nspath string) (NetNS, error) {
-	fd, err := os.Open(nspath)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to open %v: %v", nspath, err)
+const (
+	NSFS_MAGIC   = 0x6e736673
```
Comment (jonboulle) [inline pkg/ns/ns.go]: where are these from? link please

Comment (steveeJ) [inline pkg/ns/ns.go]: reference added, thanks

**Comment by jonboulle** [comment]
LGTM

**Pull request merged**